packages/lib/src/index.ts

```diff
@@ -139,3 +139,46 @@ export type RecursivePartial<T> = {
     ? RecursivePartial<U> | null
     : T[P];
 };
+
+export class IdAllocator {
+  private static registry: Array<IdAllocator> = [];
+
+  private value = 0;
```

Comment:
Since the current expectation is that the caller with pre/append a unique string, I'm thinking we can take in the prefix as an argument to the constructor. how does that sound?

Reply:
I dig that. Especially since the IDs aren't incrementing globally, that makes the prefix absolutely necessary.

Reply:
Love this, we've needed something like this for a while.